dis_snek/models/discord_objects/guild.py

```diff
@@ -662,6 +664,80 @@ async def delete_channel(self, channel: Union["TYPE_GUILD_CHANNEL", "Snowflake_T
 
         await channel.delete(reason)
 
+    async def list_scheduled_events(self, with_user_count: bool = False) -> List["ScheduledEvent"]:
+        """
+        List all scheduled events in this guild.
+
+        returns:
+            A list of scheduled events.
+        """
+        scheduled_events_data = await self._client.http.list_schedules_events(self.id, with_user_count)
+        return ScheduledEvent.from_list(scheduled_events_data, self._client)
+
+    async def get_scheduled_event(
+        self, scheduled_event_id: "Snowflake_Type", with_user_count: bool = False
+    ) -> "ScheduledEvent":
+        """
+        Get a scheduled event by id.
+
+        parameters:
+            event_id: The id of the scheduled event.
+
+        returns:
+            The scheduled event.
+        """
+        scheduled_event_data = await self._client.http.get_scheduled_event(self.id, scheduled_event_id, with_user_count)
+        return ScheduledEvent.from_dict(scheduled_event_data, self._client)
+
+    async def create_scheduled_event(
+        self,
+        name: str,
+        entity_type: "ScheduledEventEntityType",
+        scheduled_start_time: "Timestamp",
+        scheduled_end_time: Optional["Timestamp"] = MISSING,
+        description: Optional[str] = MISSING,
+        channel_id: Optional["Snowflake_Type"] = MISSING,
+        entity_metadata: Optional[dict] = MISSING,
+        privacy_level: "ScheduledEventPrivacyLevel" = ScheduledEventPrivacyLevel.GUILD_ONLY,
+        reason: Optional[str] = MISSING,
+    ):
+        """Create a scheduled guild event.
+
+        Args:
+            name: event name
+            entity_type: event type
+            start_time: `Timestamp` object
+            end_time: `Timestamp` object
+            description: event description
+            channel_id: channel id
+            entity_metadata: event metadata
+            privacy_level: event privacy level
+            reason: reason for creating this scheduled event
+
+        Returns:
+            ScheduledEvent object
+
+        !!! note
+            channel_id is Optional when entity_type is `ScheduledEventEntityType.EXTERNAL`
```

Comment:
```suggestion
            channel_id is Optional when entity_type is `ScheduledEventEntityType.EXTERNAL`
            
            location must also be supplied when this is true.
```

Reply:
Yep. Location should definitely be a first-class parameter here, rather than something hidden in entity_metadata

Reply:
This is just in case entity_metadata receives more things in the future.

Reply:
I'm not saying to remove entity_metadata.  Just allow users to specify location without it